consensus/consensus.go

```diff
@@ -34,6 +34,9 @@ type Consensus interface {
 	// GetSyncProgression retrieves the current sync progression, if any
 	GetSyncProgression() *progress.Progression
 
+	// GenerateExitProof generates exit proof for given exit event id
+	GenerateExitProof(exitID, epoch, checkpointNumber uint64) ([]types.Hash, error)
```

Comment:
Can not be done. Only an instance of consensus interface (in our case `Polybft`) has a reference to the `consensusRuntime` object, which has that state where we keep all the consensus related data. Exposing runtime to some other struct or interface is possible, but is it ok to do so? That interface would also need to reference polybft, since consensus runtime is creates in polybft when we are a validator.

Reply:
> That interface would also need to reference polybft, since consensus runtime is created in polybft when we are a validator.

Haven't entirely understood this part. We'll have a quick chat about it tomorrow.

Reply:
Anyhow, my point here is that this way, you are forcing other consensus protocols to implement `GenerateExitProof` function, although it is related only to bridge workflow.

Reply:
After some more thinking, I think that I get it: at the point where you are creating `jsonRPCHub`, you aren't aware that there is `consensusRuntime`, since we don't have some proper abstraction for it.

Possibly some quick solution would be that `Consensus` interface expose some function called `GetBridgeDataProvider` which would return `consensusRuntime` instance in the `polybft` consensus protocol (otherwise `nil`). Otherwise the rest would remain as I've proposed earlier: `consensusRuntime` implements `BridgeDataProvider` interface. That approach would help us better isolate parts of the system (if we for example one day decide to expose more bridge-related functions to the JSON RPC or any other part of the system). WDYT?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/844/commits/9b64cc29943e1df78393686cf9e9b189fc6b7ced